async-nats/src/jetstream/mod.rs

```diff
@@ -21,10 +21,64 @@ pub mod stream;
 
 pub use context::Context;
 
+/// Creates a new JetStream [Context] that provides JetStream API for managming and using [Stream],
+/// [Consumer], key value and object store.
+///
+/// # Examples:
+///
+/// ```no_run
+/// # #[tokio::main]
+/// # async fn main() -> Result<(), async_nats::Error> {
+///
+/// let nc = async_nats::connect("localhost:4222").await?;
+/// let jetstream = async_nats::jetstream::new(nc);
+///
+/// jetstream.publish("subject".to_string(), "data".into()).await?;
+/// # Ok(())
+/// # }
+/// ```
 pub fn new(client: Client) -> Context {
     Context::new(client)
 }
 
+/// Creates a new JetStream [Context] with given JetStteam domain.
+///
+/// # Examples:
+///
+/// ```no_run
+/// # #[tokio::main]
+/// # async fn main() -> Result<(), async_nats::Error> {
+///
+/// let nc = async_nats::connect("localhost:4222").await?;
+/// let jetstream = async_nats::jetstream::with_domain(nc, "hub");
```

Comment:
Same, lets start aiming for consistency in all examples:

Personally leaning towards just naming them the types they are:
```suggestion
/// let context = async_nats::jetstream::with_domain(nc, "hub");
```